packages/core/src/walk.ts

```diff
@@ -329,6 +329,11 @@ export function walkDocument<T>(opts: {
       ruleId: string,
       severity: ProblemSeverity,
     ) {
+      if ((node === null || node === undefined) && (context.type?.name !== 'scalar')) {
+        // if node is not a scalar then visitor may crash when null or undefined is passed
+        // instead of a node when it tries to access any fields
+        return;
```

Comment:
If we do just this, we miss visitor run for null values so we won't be able to show validation errors:
Try changing any field (e.g. `Info.descrioption`) in the valid openapi to null, you won't get an error.